x/vote/keeper/msg_server.go

```diff
@@ -63,6 +63,10 @@ func (s msgServer) Vote(c context.Context, req *types.VoteRequest) (*types.VoteR
 		return &types.VoteResponse{Log: fmt.Sprintf("poll %s failed", poll.GetKey())}, nil
 	case poll.Is(vote.Expired):
 		return &types.VoteResponse{Log: fmt.Sprintf("poll %s expired", poll.GetKey())}, nil
+	case poll.Is(vote.Completed):
+		event = event.AppendAttributes(sdk.NewAttribute(types.AttributeKeyPollState, vote.Completed.String()))
+
+		fallthrough
```

Comment:
You can't fallthrough here. `fallthrough` in golang blindly goes to execute the logic in the next case without checking the case condition at all. This would cause the poll result to be handled multiple times.